cluster/images/etcd/migrate/versions.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// EtcdVersion specifies an etcd server binaries SemVer.
+type EtcdVersion struct {
+	major uint64
+	minor uint64
+	patch uint64
+}
+
+// ParseEtcdVersion parses a SemVer string to an EtcdVersion.
```

Comment:
This does not support SemVer pre-release versions or release versions with build metadata, which may be ok?

Reply:
I'll swap over to a SemVer library like @lavalamp suggested.